File: server/models/Meeting.js
const mongoose = require('mongoose');

const meetingSchema = new mongoose.Schema({
  title: { type: String, default: '' },
  status: { type: String, enum: ['active', 'ended'], default: 'active' },
  rawTranscript: { type: String, default: '' },
  cleanedTranscript: { type: String, default: '' },
  summary: [{ type: String }],
  actionItems: [{ type: String }],
  rollingSnapshot: { type: String, default: '' },
  lastSummaryAt: { type: Date },
  emailSentTo: { type: String, default: '' },
  startedAt: { type: Date, default: Date.now },
  endedAt: { type: Date },
}, { timestamps: true });

module.exports = mongoose.model('Meeting', meetingSchema);

File: ios/MeetingTranscriber/Models/Meeting.swift
import Foundation

struct Meeting: Identifiable, Codable, Hashable {
    let id: String
    var title: String
    var status: MeetingStatus
    var rawTranscript: String
    var cleanedTranscript: String
    var summary: [String]
    var actionItems: [String]
    var rollingSnapshot: String
    var startedAt: Date
    var endedAt: Date?

    enum MeetingStatus: String, Codable {
        case active, ended
    }

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case title, status, rawTranscript, cleanedTranscript
        case summary, actionItems, rollingSnapshot, startedAt, endedAt
    }

    var displayTitle: String { title.isEmpty ? "Untitled Meeting" : title }
}

struct MeetingListItem: Identifiable, Codable {
    let id: String
    var title: String
    var status: Meeting.MeetingStatus
    var startedAt: Date
    var endedAt: Date?

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case title, status, startedAt, endedAt
    }

    var displayTitle: String { title.isEmpty ? "Untitled Meeting" : title }
}
